src/Symfony/Component/Form/Form.php

```diff
@@ -580,6 +580,8 @@ public function submit($submittedData, $clearMissing = true)
                         if (method_exists($child, 'getClickedButton') && null !== $child->getClickedButton()) {
                             $this->clickedButton = $child->getClickedButton();
                         }
+                    } elseif (!$isSubmitted && !$clearMissing && array_key_exists($name, $submittedData)) {
```

Comment:
This never evaluates to TRUE, because $isSubmitted = array_key_exists($name, $submittedData);